3-Register/src/app/signed-in/[username]/page.tsx

```diff
@@ -0,0 +1,37 @@
+import { CelebrationIcon } from "@/ui/Icons";
+import { Ubuntu } from "next/font/google";
+
+const ubuntu = Ubuntu({ subsets: ["latin"], weight: ["500"] });
+interface SignedInProps {
+  params: {
+    username: string;
+  };
+}
+
+const simulateDelay = async (miliseconds: number) => {
+  return await new Promise((resolve) => setTimeout(resolve, miliseconds));
+};
+
+export default async function SignedIn({
+  params: { username },
+}: SignedInProps) {
+  await simulateDelay(3000);
```

Comment:
I've never seen an `async` component, is that a `next.js` thing?

Reply:
Yes! It's a feature of Next. All components are React server components by default, and they can be asynchronous. This is very helpful if you're doing some data fetching because you can directly call an async function that fetches the data in the component. In this case, while we're awaiting the call to `simulateDelay`, next will render `loading.tsx` instead of `pages.tsx`.